src/components/core/PivotTable.tsx

```diff
@@ -315,34 +314,39 @@ export const RowLoadingElement = (props: ICellRendererParams) => {
 };
 
 export const getDrillIntersection = (
-    drillItems: IDrillItem[],
+    drillItems: IMappingHeader[],
     afm: AFM.IAfm
 ): IDrillEventIntersectionElement[] => {
     // Drilling needs refactoring: all '' should be replaced by null (breaking change)
     // intersection consists of
         // 0..1 measure
         // 0..1 row attribute and row attribute value
         // 0..n column attribute and column attribute values
-    return drillItems.map((drillItem: IDrillItem) => {
-        const { identifier, uri, title } = drillItem;
+    return drillItems.map((drillItem: IMappingHeader) => {
+        let headerLocalIdentifier = null;
+        let headerIdentifier = '';
+        let uriAndIdentifier = null;
+
+        if (!isMappingHeaderAttributeItem(drillItem)) {
+            headerLocalIdentifier = getMappingHeaderLocalIdentifier(drillItem);
+            headerIdentifier = getMappingHeaderIndentifier(drillItem) || '';
+            uriAndIdentifier = headerLocalIdentifier
+                ? getMasterOrDerivedUriAndIdentifier(afm, headerLocalIdentifier)
+                : null;
+        }
+
+        const headerUri = getMappingHeaderUri(drillItem) || '';
+        const uri = uriAndIdentifier && uriAndIdentifier.uri || headerUri;
+        const identifier = uriAndIdentifier && uriAndIdentifier.identifier || headerIdentifier;
+        const id = headerLocalIdentifier || headerIdentifier;
```

Comment:
I'm not sure about this, but maybe we could have a helper method like getIdFromDrillItem() which would contain all these extractions which are on above lines.

Reply:
I understand. Currently, we have at least 4 places where drill intersection is created. But unifying this is pretty complex to do it now. I have created followup https://jira.intgdc.com/browse/BB-1318 which should unify this.